From cf896b6b81b6ade4b0c2fa16bdd3fe281de3d8cf Mon Sep 17 00:00:00 2001
From: jie shen <jie.shen@unisoc.com>
Date: Wed, 29 Jul 2020 10:27:51 +0800
Subject: [PATCH 28/47] Add isFaceAuthEnabledForUser checks in
 KeyguardLiftController

When request pick gesture sensor, judge that if the current user does not turn on the face unlocking function, the sensor should not be Listening

Fixes: 161954958
Test: presubmit

Change-Id: I34552315049e07f123403fe693251edb460fef5d
Signed-off-by: SamarV-121 <samarvispute121@gmail.com>
---
 .../systemui/statusbar/phone/KeyguardLiftController.kt      | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardLiftController.kt b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardLiftController.kt
index 74d9f540..bfe0684b 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardLiftController.kt
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardLiftController.kt
@@ -84,7 +84,9 @@ class KeyguardLiftController constructor(
         val onKeyguard = keyguardUpdateMonitor.isKeyguardVisible &&
                 !statusBarStateController.isDozing
 
-        val shouldListen = onKeyguard || bouncerVisible
+        val userId = KeyguardUpdateMonitor.getCurrentUser()
+        val isFaceEnabled = keyguardUpdateMonitor.isFaceAuthEnabledForUser(userId)
+        val shouldListen = (onKeyguard || bouncerVisible) && isFaceEnabled
         if (shouldListen != isListening) {
             isListening = shouldListen
 
@@ -95,4 +97,4 @@ class KeyguardLiftController constructor(
             }
         }
     }
-}
\ No newline at end of file
+}
-- 
2.30.1

